mindocr/data/transforms/det_transforms.py

```diff
@@ -183,3 +186,203 @@ def expand_poly(poly, distance: float, joint_type=pyclipper.JT_ROUND) -> List[li
+    return peri
+
+
+def shrink(text_polys, rate, max_shr=20):
+    rate = rate * rate
+    shrinked_text_polys = []
+    for bbox in text_polys:
+        area = Polygon(bbox).area
+        peri = perimeter(bbox)
+
+        pco = pyclipper.PyclipperOffset()
+        pco.AddPath(bbox, pyclipper.JT_ROUND, pyclipper.ET_CLOSEDPOLYGON)
+        offset = min((int)(area * (1 - rate) / (peri + 0.001) + 0.5), max_shr)
+
+        shrinked_bbox = pco.Execute(-offset)  # (N, 2) shape, N maybe larger than or smaller than 4.
+        if not shrinked_bbox:
+            shrinked_text_polys.append(bbox)
+            continue
+
+        shrinked_bbox = np.array(shrinked_bbox)[0]
+        if shrinked_bbox.shape[0] <= 2:
+            shrinked_text_polys.append(bbox)
+            continue
+
+        shrinked_text_polys.append(shrinked_bbox)
+
+    return shrinked_text_polys
+
+
+class PSEGtDecode(object):
+    def __init__(self, kernel_num=7, min_shrink_ratio=0.4, min_shortest_edge=640):
+        self.kernel_num = kernel_num
+        self.min_shrink_ratio = min_shrink_ratio
+        self.min_shortest_edge = min_shortest_edge
+
+    def __call__(self, data):
+
+        image = data['image']
+        text_polys = data['polys']
+        ignore_tags = data['ignore_tags']
+
+        h, w, _ = image.shape
+        short_edge = min(h, w)
+        if short_edge < self.min_shortest_edge:
+            # keep short_size >= self.min_short_edge
+            scale = self.min_shortest_edge / short_edge
+            image = cv2.resize(image, dsize=None, fx=scale, fy=scale)
+            text_polys *= scale
+
+        # get gt_text and training_mask
+        img_h, img_w = image.shape[0: 2]
+        gt_text = np.zeros((img_h, img_w), dtype=np.float32)
+        training_mask = np.ones((img_h, img_w), dtype=np.float32)
+        if text_polys.shape[0] > 0:
+            text_polys = text_polys.astype('int32')
+            for i in range(text_polys.shape[0]):
+                cv2.drawContours(gt_text, [text_polys[i]], 0, i + 1, -1)
+                if ignore_tags[i]:
+                    cv2.drawContours(training_mask, [text_polys[i]], 0, 0, -1)
+
+        # get gt_kernels
+        gt_kernels = []
+        for i in range(1, self.kernel_num):
+            rate = 1.0 - (1.0 - self.min_shrink_ratio) / (self.kernel_num - 1) * i
+            gt_kernel = np.zeros((img_h, img_w), dtype=np.float32)
+            kernel_text_polys = shrink(text_polys, rate)
+            for j in range(len(kernel_text_polys)):
+                cv2.drawContours(gt_kernel, [kernel_text_polys[j]], 0, 1, -1)
+            gt_kernels.append(gt_kernel)
+
+        gt_text[gt_text > 0] = 1
+        gt_kernels = np.array(gt_kernels)
+
+        data['image'] = image
+        data['polys'] = text_polys
+        data['gt_kernels'] = gt_kernels
+        data['gt_text'] = gt_text
+        data['mask'] = training_mask
+        return data
+
+
+class DetResizeForTest(object):
```

Comment:
Replace this operation with DetResize added in https://github.com/mindspore-lab/mindocr/pull/294
and set the args in yaml config as follows, which should be equivalent to this DetResizeForTest op.
```
DetResize:
          target_limit_side: 1472
          limit_type: min
```

Please verify it by running the evaluation script

Reply:
Already replaced and verified.